linchpin/__init__.py

```diff
@@ -688,6 +709,14 @@ def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
 
             if provision_data[target].get('layout', None):
                 l_data = provision_data[target]['layout']
+                try:
+                    self._validate_layout(l_data)
+                except SchemaError as e:
+                    print e
+                    print "\n"
```

Comment:
These two print statements should either be in a ctx.log_state, or managed somehow else.